client/src/c_console.cpp

```diff
@@ -418,6 +448,84 @@ void ConsoleCommandLine::deleteCharacter()
 	}
 }
 
+
+void ConsoleCommandLine::deleteLeftWord()
+{
+	size_t word = 0;
+	bool firstSpacesCleared = false;
+	bool spaceWord = false;
+
+	const char* str = text.c_str();
+
+	// Delete trailing spaces instead of the word
+	if (str[cursor_position - 1] == ' ')
+	{
+		spaceWord = true;
+	}
+
+	while (cursor_position > 0)
+	{
+		if (spaceWord)
+		{
+			if (str[cursor_position - 1] != ' ')
+			{
+				break;
+			}
+		}
+		else
+		{
+			if (firstSpacesCleared && str[cursor_position] == ' ')
+			{
+				break;
+			}
+			else if (str[cursor_position] != ' ' && !firstSpacesCleared)
+			{
+				firstSpacesCleared = true;
+			}
+		}
+
+		cursor_position--;
+		word++;
+	}
+
+	text.erase(cursor_position, word);
+
+	doScrolling();
+}
+
+void ConsoleCommandLine::deleteRightWord()
+{
+	bool spaceWord = false;
+
+	const char* str = text.c_str();
+
+	if (str[cursor_position] == ' ')
```

Comment:
What happens if `cursor_position` is equal to the length of the string?

Reply:
Changed to use `isspace()` and check `cursor_position` for all of these.

Reply:
Also check bounds here to make sure `cursor_position` is less than `text.length()`